flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkResourceManager.java

```diff
@@ -78,6 +79,9 @@
 	/** The containers where a TaskManager is starting and we are waiting for it to register */
 	private final Map<ResourceID, YarnContainerInLaunch> containersInLaunch;
 
+	/** The container where a TaskManager has been started and is running in */
+	private final Map<ResourceID, Container> containersLaunched;
```

Comment:
It is true that it holds the registered resources but it does not hold the launched containers. When a `JobManager` loses its leadership the list of registered workers will be cleared. In order to reconstruct the mapping `ResourceID --> Container`, you need this new map.